src/hooks/useFetchImages.ts

```diff
@@ -1,71 +1,78 @@
   const [endDate, setEndDate] = useState(unixToDate(unixNow))
 
-  // Updating time frames once week of images has been called
-  useEffect(() => {
-    setEndDate(() => {
-      const startDateAsUnix = dateToUnix(startDate)
-      const dayBeforePrevStart = unixToDate(startDateAsUnix - 43200)
-
-      return dayBeforePrevStart
-    })
-
-    setStartDate((prevStart) => {
-      const prevStartAsUnix = dateToUnix(prevStart)
-      const sevenDaysFromStart = unixToDate(prevStartAsUnix - 518400)
-
-      return sevenDaysFromStart
-    })
-  }, [increment, isLoading])
-
-  // Api Call
-  useEffect(() => {
-    if (!needMoreImages) return
-
-    const requestApiData = async () => {
+  const {
+    isLoading,
+    error,
+    data: images,
+  } = useQuery({
+    enabled: needMoreImages,
+    queryKey: ['images'],
+    queryFn: async () => {
       setIncrement((count) => (count += 1))
-      setIsLoading(true)
-
-      // @ts-expect-error
-      const env = import.meta.env.VITE_NASA_API_KEY
-      const URL = `https://api.nasa.gov/planetary/apod` // ?api_key=${env}&start_date=${startDate}&end_date=${endDate}
 
       try {
+        // @ts-expect-error
+        const ENV = import.meta.env.VITE_NASA_API_KEY
+        const URL = `https://api.nasa.gov/planetary/apod`
         const request = await ky.get(URL, {
           retry: 3,
           searchParams: {
-            api_key: env,
+            api_key: ENV,
             start_date: startDate,
             end_date: endDate,
           },
         })
+
         if (request.status === 200) {
-          // reversing so today's image shows first
-          const data: Image[] = await request.json()
-          data.reverse()
+          const nasaData: Image[] = await request.json()
 
-          setImages((prevImages) => [...prevImages, ...data])
-          setIsLoading(false)
-        } else console.error('Failed with status code: ', request.status, request.statusText)
+          return nasaData.reverse() // reversing so today's image shows first
+        } else {
+          console.error('Failed with status code: ', request.status, request.statusText)
+        }
       } catch (err) {
         console.error(err)
       }
-    }
+    },
+  })
+
+  // Updating time frames once week's worth of images have been called
+  useEffect(() => {
+    setEndDate(() => {
+      const startDateAsUnix = dateToUnix(startDate)
+      const dayBeforePrevStart = unixToDate(startDateAsUnix - 43200)
+
+      return dayBeforePrevStart
+    })
+
+    setStartDate((prevStart) => {
+      const prevStartAsUnix = dateToUnix(prevStart)
+      const sevenDaysFromStart = unixToDate(prevStartAsUnix - 518400)
+
+      return sevenDaysFromStart
+    })
+  }, [increment, isLoading])
 
-    requestApiData()
-  }, [needMoreImages])
+  isLoading ? { isLoading: true, data: images || [] } : null
+  error ? { isLoading: false, data: images || [] } : null
 
-  return { images, isLoading }
+  return { isLoading, data: images || [] }
```

Comment:
Now we can return the entire result of useQuery
```suggestion
  return nasaImagesQuery
```